static/app/components/events/featureFlags/eventFeatureFlagList.tsx

```diff
@@ -0,0 +1,160 @@
+import {useCallback, useRef, useState} from 'react';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import DropdownButton from 'sentry/components/dropdownButton';
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
+import {
+  CardContainer,
+  FeatureFlagDrawer,
+  FLAG_SORT_OPTIONS,
+  FlagSort,
+  getLabel,
+} from 'sentry/components/events/featureFlags/featureFlagDrawer';
+import useDrawer from 'sentry/components/globalDrawer';
+import KeyValueData, {
+  type KeyValueDataContentProps,
+} from 'sentry/components/keyValueData';
+import {IconChevron, IconSort} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import type {Event, FeatureFlag} from 'sentry/types/event';
+import type {Group} from 'sentry/types/group';
+import type {Project} from 'sentry/types/project';
+
+export function EventFeatureFlagList({
+  event,
+  group,
+  project,
+}: {
+  event: Event;
+  group: Group;
+  project: Project;
+}) {
+  const [isCollapsed, setIsCollapsed] = useState(false);
+  const [sortMethod, setSortMethod] = useState<FlagSort>(FlagSort.EVAL);
+  const {closeDrawer, isDrawerOpen, openDrawer} = useDrawer();
+  const viewAllButtonRef = useRef<HTMLButtonElement>(null);
+
+  // transform the flags array into something readable by the key-value component
+  const hydrateFlags = (flags: FeatureFlag[] | undefined) => {
+    if (!flags) {
+      return [];
+    }
+    return flags.map(f => {
+      return {item: {key: f.flag, subject: f.flag, value: f.result}};
+    });
+  };
+
+  const initialFlags = hydrateFlags(event.contexts?.flags?.values);
+  const [flags, setFlags] = useState<KeyValueDataContentProps[]>(initialFlags);
```

Comment:
It's rare i see a need for this, but `useState` can accept an initializer function which will only be called only once. In this case it'll save the mapper from being run on each render.

```suggestion
  const [flags, setFlags] = useState<KeyValueDataContentProps[]>(() => hydrateFlags(event.contexts?.flags?.values));
```

https://react.dev/reference/react/useState#usestate

Reply:
i wanted to have a const bc we reuse `initialFlags` in a bunch of spots after - we pass it to the drawer & we record it so we can reset the sort with the original ordering